### PR TITLE
Brackets checking removed from the return condition.

### PR DESCRIPTION
**Pull request by floor12**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  |❌
| Breaks BC?    | ❌
| Tests pass?   | ❌
| Fixed issues  | #18001

Let watch how the tests falls.

### CODE REVIEW

**Comment by samdark** [comment]
Seems to fail tests. At least table quoting ones.

**Comment by floor12** [comment]
I updated the code to check brackets in first and last position in table name string. When all tests passed I also added new test to check quoting table name with brackets.

**Review by samdark — APPROVED** [review]
Please add a line for CHANGELOG.md

**Comment by floor12** [comment]
@samdark Sorry for my stupid question, but its my first contribute in a such big project. So, my question is about CHANGELOG. I made all this changes from the last stable release 2.0.35  (because master branch had test fails at the moment of forking). So, my version of CHANGELOG hasn't list of 2.0.36 changes. And im not sure that it will be merged properly if I just add new line. So, I should merge master into my branch, add the line with my changes, and make new pull request, o something else?
Thanks.

File: framework/db/Schema.php [hunk 1]
```diff
@@ -478,7 +478,11 @@ public function quoteValue($str)
      */
     public function quoteTableName($name)
     {
-        if (strpos($name, '(') !== false || strpos($name, '{{') !== false) {
+
+        if (strpos($name, '(') === 0 && strpos($name, ')') === strlen($name) - 1) {
```
Comment (rob006) [inline framework/db/Schema.php:482]:
```suggestion
        if (strncmp($name, '(', 1) === 0 && substr($name, -1) === ')') {
```

Comment (floor12) [reply framework/db/Schema.php:482]: @rob006 I made a few speed tests and found that your variant works slowly for 5-15%. I was surprised but it's true.

**Comment by samdark** [comment]
@floor12 I'll handle it, don't worry.

File: framework/db/Schema.php (hunk 1 above)
Comment (rob006) [reply framework/db/Schema.php:482]: Hmm, `strpos()` indeed could be more efficient in case of short strings.

**Comment by samdark** [comment]
Merged. Thank you.

**Pull request merged**